src/azure-cli/azure/cli/command_modules/network/_template_builder.py

```diff
@@ -172,6 +183,17 @@ def _ag_subresource_id(_type, name):
     }
     if cmd.supported_api_version(min_api='2018-08-01'):
         ag.update({'zones': zones})
+    if user_assigned_identity and cmd.supported_api_version(min_api='2018-12-01'):
+        ag.update(
+            {
+                "identity": {
+                    "type": "UserAssigned",
+                    "userAssignedIdentities": {
+                        user_assigned_identity: {}
```

Comment:
Why do you update user_assigned_identity to {} here not the userAssignedIdentity info?